### DISCUSSION TITLE
auto turn on/off and start long term timelapse

### DISCUSSION THREAD

**Discussion by CptCurk** [gopro/labs#579] [Q&A] [answered]
Hello, I'm trying to make a "long terme timelapse" (turning of between the photos) shooting between 9am and 5pm, with a photo each 2'30'', I'm not adding any special stuff like lenses there. 

**First :**
I've seen that post : https://github.com/gopro/labs/discussions/524 but I don't understand everything in the QR here : https://gopro.github.io/labs/control/set/?cmd=!18:00NmTPfW0g0v0q0dToV4B1D0R1C1S1oT230911203548!S!06:00E!56NQmPN!S!1R

!18:00N - start time 6pm
mTP - mode timelapse photo
fW - wide lense

then this... I don't find answer - 0g0v0q0dToV4B1D0R1C1S1oT230911203548 
0g0v0q0dT I don't understand that... 
oTyymmddhhmmss.sss this is for time and date okay. But why ? 
oV4B1D0R1C1S1 is this volume, brightness, LED & LCD... ? So we don't need to add everytime the o in front of it ? 

!S!06:00E mean to start now, end at 6am. Why not only !06:00E ?

!56NQmPN!S!1R this last sequence is the long term timelapse, I understand that. Bu why again a !S

**Second :**
my stuff 
>09:00<17:00!150NQmPN!S!1R~!O
I don't really understand with my camera if it's okay... But how would you do it ? 

Thank you for clarifying the stuff.. As you can see, I looked for it. But I need some hint to get it right in my head ! 
Best regards :-D

**Comment by dnewman-gpsw** [comment, marked as answer]
The user you copying this from was setting everything and making errors at the some time. Not a good source to learn from.  But all the command are documented https://gopro.github.io/labs/control/tech/

Start simple, and think of the operation you as a regular user would do.  Would you set the mode before every photo, or simple have camera remember what mode it was in?

mPN - Mode photo night 
dL - default Last used.

So you could have a QR Code just for initialization (or use the menu) with: **mPNdL**

The repeat part you've worked out   !150NQ!S!1R  this also works   !150SQ!1R    (command: every ~150 second press start)

Adding the condition time 
**>09:00<17:00!150SQ+!1R~!09:00R**

This translates to:
```
if(time > 09:00 AND time <17:00)  
{
   Sleep(150);
   Start();
   Sleep(1);
   Repeat();
}
else
{
  Sleep(9am);
  Repeat();
}
```

The only weird part of the language is the need of '+' between command within a conditional.

**Reply by CptCurk** [reply]
hooo that's a great answer. I indeed haven't found the page you linked there. But everything I missed was on it.
Okay. Initializing is making it easier, but double scan is a way to forgot it in the future and make a professionnal mistake. 

So in that case, to group all in one line (because in fact, I would check my menu right before shooting 😁) : mPNdL>09:00<17:00!150SQ+!1R~!09:00R
If I don't add the repeat the command at 9am, but just turn off as I did with "!O", this won't start automatically the day after at 09:00 ? 

thank you a lot again for your answer !

**Reply by dnewman-gpsw** [reply]
!O will not start the day. It is a complete shutdown, with no wake timer. It is not recommended.

mPNdL>09:00<17:00!150SQ+!1R~!09:00R    will work as a single QR.

**Reply by CptCurk** [reply]
perfect, completly answered to this issue. 
thanks a lot @dnewman-gpsw  !
